.github/workflows/build-package.yml

```diff
@@ -21,93 +14,135 @@ on:
         description: "MSYS2 packages branch to build from"
         type: string
         default: "woarm64"
+      runner_arch:
+        description: "Architecture to build on"
+        type: string
+        default: "x86_64"
+      build_with_native:
+        description: "Build with native MinGW toolchain"
+        type: boolean
+        default: false
+      needs:
+        description: "Parent workflow job dependencies"
+        type: string
+      dependencies:
+        description: "Install additional dependencies"
+        type: string
+        default: ""
 
 defaults:
   run:
     shell: msys2 {0}
 
 env:
   GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  FLAVOR: ${{ (inputs.runner_arch == 'aarch64') && (inputs.build_with_native && 'NATIVE_WITH_NATIVE' || 'NATIVE_WITH_CROSS') || 'CROSS' }}
+  CLEAN_BUILD: 1
 
 jobs:
   build:
     name: Build ${{ inputs.package_name }}
-    runs-on: windows-latest
+    timeout-minutes: 720
+    runs-on: >-
+      ${{ fromJson(inputs.runner_arch == 'aarch64'
+        && '["Windows", "ARM64", "MSYS2"]'
+        || '["windows-latest"]') }}
 
     steps:
-      - uses: msys2/setup-msys2@v2
+      - name: Kill hanging processes
+        if: inputs.runner_arch == 'aarch64'
+        shell: powershell
+        run: |
+          taskkill /F /FI 'MODULES eq msys-2.0.dll'
```

Comment:
The job has just been started, why is it hanging?

Reply:
The job is using self-hosted non-ephemeral runners. This was particularly needed when there was the x64 Cygwin on Arm64 hanging issue. It is resolved now but handling of previous processes can still happen.